smithy-swift-codegen/src/main/kotlin/software/amazon/smithy/swift/codegen/config/ClientConfiguration.kt

```diff
@@ -0,0 +1,22 @@
+package software.amazon.smithy.swift.codegen.config
+
+import software.amazon.smithy.swift.codegen.SwiftWriter
+import software.amazon.smithy.swift.codegen.integration.ServiceConfig
+
+/**
+ * Specifies the behaviour of the service configuration
+ */
+interface ClientConfiguration {
+    /**
+     * The protocol name of the client configuration
+     */
+    val swiftProtocolName: String
+
+    /**
+     *
+     */
+    fun protocolImplementation(writer: SwiftWriter, serviceConfig: ServiceConfig) {
+    }
+    fun addImport(writer: SwiftWriter) {
+    }
```

Comment:
Is there a way to encapsulate the protocol name and imports / dependencies into a `Symbol`?

Reply:
Updated `swiftProtocolName` type to `Symbol`